include/GafferScene/BranchCreator.h

```diff
@@ -195,17 +218,17 @@ class GAFFERSCENE_API BranchCreator : public FilteredSceneProcessor
 
 		/// Used to calculate the name remapping needed to prevent name clashes with
 		/// the existing scene. Must be evaluated in a context where "scene:path" is
-		/// one of the parent paths.
+		/// one of the destination paths. This is computed separately from `branchesPlug()`
+		/// so that we can defer branch evaluation as late as possible.
```

Comment:
I'm not sure what this means ... hashMapping calls branches() unconditionally, so there doesn't seem to be any deferral of branch evaluation happening?

Reply:
It's not `branches()` that we're trying to defer, but any calls to `computeBranch*()` or `hashBranch*()`. I can see that this isn't clear though.

Reply:
I did eventually figure out what this means, but yeah, it was the opposite of what I thought it was implying - it's not that mappingPlug is computed separately so that it can be faster, but that mappingPlug has to do extra calculations, which we don't want to make branchesPlug depend on.  Makes sense, but yeah, not clear from the comment.

Reply:
Clarified in https://github.com/GafferHQ/gaffer/pull/4200/commits/22df375f13dd3ddf10e2e429da957a93d7bcb555.